File: backend/app/schemas/schedule.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict


class ScheduleCreateRequest(BaseModel):
    appliance_id: int
    action: Literal["ON", "OFF"]
    scheduled_time: datetime

class ScheduleResponse(BaseModel):
    id: int
    appliance_id: int
    action: str
    scheduled_time: datetime
    created_by: int | None = None
    created_at: datetime
    executed_at: datetime | None = None

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/schedule.ts
export interface ScheduleCreateRequest {
  appliance_id: number
  action: "ON" | "OFF"
  scheduled_time: string
}

export interface Schedule {
  id: number
  appliance_id: number
  action: string
  scheduled_time: string
  created_by: number | null
  created_at: string
  executed_at: string | null
}
